python-sdk/src/astro/sql/operators/data_validations/check_table.py

```diff
@@ -44,13 +44,17 @@ def __init__(
         task_id: Optional[str] = None,
         **kwargs,
     ):
+        # We need astro table object to get conn_id from it. Adding to template_fields doesn't work for __init__()
+        table = dataset
+        if not isinstance(table, BaseTable):
+            table = dataset.operator.output_table
 
-        db = create_database(dataset.conn_id)
+        db = create_database(table.conn_id)
         super().__init__(
-            table=db.get_table_qualified_name(dataset),
+            table=db.get_table_qualified_name(table),
             checks=checks,
             partition_clause=partition_clause,
-            conn_id=dataset.conn_id,
+            conn_id=table.conn_id,
```

Comment:
@utkarsharma2 you shouldn't be creating databases in the init function of a class. This will run every single time that the airflow scheduler parses the DAG. You should override the execute function and add these elements before running "super.execute()".

Reply:
@dimberman I thought of doing that but then there was a risk since I'll be assuming the variable name used by the Parent operator like `self.table` and `self.conn_id` if the parent function changed in these variables it will cause a failure in check_table operator. 

Also, the query is generated in `__init__` of SQLTableCheckOperator. So we also have regenerate that in execute() of check_table().

Reply:
Moved the logic to execute()